addons/website_event/controllers/main.py

```diff
@@ -123,14 +123,15 @@ def events(self, page=1, **searches):
             'current_type': current_type,
             'event_ids': events,  # event_ids used in website_event_track so we keep name as it is
             'dates': dates,
-            'categories': request.env['event.tag.category'].search([('is_published', '=', True)]),
+            'categories': request.env['event.tag.category'].search([('is_published', '=', True)]).filtered(lambda r: website.id==r.website_id.id or r.website_id.id==False),
```

Comment:
Can't we use a stricter domain rather than filtering right after searching?

Reply:
@thiryr can you explain little bit more? what kind of stricter domain?

Reply:
`[('is_published', '=', True), '|', ('website_id', '=', website.id), ('website_id', '=', False)]` kind of thing, since the field is stored it's faster to get psql to do the filtering for us

Reply:
@thiryr i will try, makes lot of sense